api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+/**
+ * @swagger
+ * tags:
+ *   name: Ranks
+ *   description: Rank Management
+ */
+
+import express, { Request, Response } from "express";
+import logger from "../utilities/logger";
+import pool from "../database/index";
+import { Rank } from "../utilities/types";
+import { indexOf } from "lodash";
+import { paginationIsValid } from "../utilities/validation";
+
+const router = express.Router();
+
+const getGameRanks = async (id: number) => {
+  try {
+    const result = await pool.query("SELECT * FROM games WHERE game_id = $1", [
+      id,
+    ]);
+    return result.rows[0].rank_types;
+  } catch (error) {
+    logger.error(error);
+  }
+};
+
+const getAverageRank = async (ranks: Rank[]) => {
+  let average = 0;
+  if (ranks[0].numeric_rank) {
+    ranks.forEach((rank) => (average += rank.numeric_rank));
+    return average / ranks.length;
+  } else {
+    const gameRanks = await getGameRanks(ranks[0].game_id);
+    ranks.forEach((rank) => {
+      average += indexOf(gameRanks, rank.text_rank);
+    });
+    return gameRanks[average / ranks.length];
+  }
+};
+
+/**
+ * @swagger
+ * /ranks:
+ *   get:
+ *     summary: Get all ranks
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks of all users from the database.
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+router.get("/", async (req: Request, res: Response) => {
```

Comment:
This can remain in this file, but should we also have a /{id} endpoint to get a single entry?

Reply:
Although I don't think it's necessary, I'll add it in for debugging purposes.